File: leetcode/golang/1.两数之和.go
func twoSum(nums []int, target int) []int {
    d, ret := make(map[int]int), make([]int, 2);
    for i, num := range nums {
        if _, ok := d[target - num];!ok {
            d[num] = i;
        } else {
            ret[0] = d[target - num];
            ret[1] = i;
            return ret;
        }
    }
    return ret;
}

File: leetcode/python/1.两数之和.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d, ret = dict(), list()
        for i in range(len(nums)):
            if target - nums[i] not in d:
                d[nums[i]] = i
            else:
                ret.append(d[target - nums[i]])
                ret.append(i)
                return ret
        return ret